includes/blocks/class-sensei-blocks.php

```diff
@@ -178,4 +178,51 @@ public function has_sensei_blocks( $post = null ) {
 
 		return false !== strpos( (string) $post, '<!-- wp:sensei-lms/' );
 	}
+
+	/**
+	 * Update the URL of a button block.
+	 *
+	 * @param string $block_content The block content about to be appended.
+	 * @param array  $block         The full block, including name and attributes.
+	 * @param string $class_name    The CSS class name used to identify the correct block.
+	 * @param string $url           The URL to navigate to when the button is clicked.
+	 *
+	 * @return string Block HTML.
+	 */
+	public static function update_button_block_url( $block_content, $block, $class_name, $url ): string {
+		if ( ! isset( $block['blockName'] ) || 'core/button' !== $block['blockName'] ) {
```

Comment:
Do we have the `"more-courses"` className attribute in the `$block` object to also check? Just to avoid running it for all buttons.

Reply:
@donnapep, just added this commit with this change: https://github.com/Automattic/sensei/pull/4266/commits/1de4577f4250ae72493a3ec984efb6a4a5755161

I also noticed that we didn't need to loop through the attributes since we had that information in the block attributes. I'm already merging it, but any problem you notice, feel free to reach me out to fix it.